serverlessworkflow/modules/ROOT/pages/security/authention-support-for-openapi-services.adoc

```diff
@@ -1 +1,359 @@
+quarkus.oidc-client.oauth_example.grant.type=client
+quarkus.oidc-client.oauth_example.credentials.client-secret.method=basic
+quarkus.oidc-client.oauth_example.credentials.client-secret.value=secret
+----
+
+The configuration suffix `quarkus.oidc-client.oauth_example` is exclusive for the schema defined in the specification file and the scheme name is sanitized by applying the rules described <<sanitize_note, above>>.
+
+For this to work you **must** add the https://quarkus.io/guides/security-openid-connect-client#oidc-client-filter[Quarkus OIDC Client Filter Extension] to your project:
+
+[source, xml]
+----
+<dependency>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-oidc-client-filter</artifactId>
+</dependency>
+----
+
+== Authorization Token Propagation
+
+The authorization token propagation can be used with OpenAPI operations that are secured with a security scheme of type `oauth2` or `http bearer`.
+When configured, you can propagate the authorization tokens passed to your Serverless Workflow at the time it was created, along the OpenAPI invocations executed by that service.
+These propagations must be configured individually for each of the corresponding security schemes, in that way, you can do a fine-tuned configuration of which invocations require the propagation.
+Similarly to the other security scheme configurations, the token propagation apply to all the operations secured with the same scheme.
+
+Following the <<_oauth2_authentication>> example presented above, these are the supported configurations:
+
+[frame=all]
+|===
+|Property Key |Example
+
+| `quarkus.openapi-generator.[filename].auth.[security_scheme_name].token-propagation=[true,false]`
+| `quarkus.openapi-generator.security_example_json.auth.oauth_example.token-propagation=true` +
+ +
+Enables the token propagation for all the operations that are secured with the `oauth-example` scheme in the `security-example.json` file. By default, the standard HTTP Authorization header is propagated.
+
+| `quarkus.openapi-generator.[filename].auth.[security_scheme_name].header-name=[http_header_name]`
+| `quarkus.openapi-generator.security_example_json.auth.oauth_example.header-name=MyHeaderName` +
+ +
+Stays that the authorization token to propagate will be read from the HTTP header `MyHeaderName` instead of the standard HTTP `Authorization` header.
+
+|===
+
+[IMPORTANT]
+====
+The authorization tokens are propagated as long the Serverless Workflow doesn't reach a "waiting" state, for example to wait for an event to arrive. When the Serverless Workflow is resumed the tokens will be no longer propagated.
+====
+
+== Going further
+
+The Serverless Workflows OpenAPI support is based on the https://github.com/quarkiverse/quarkus-openapi-generator[Quarkus OpenAPI Generator] extension. While not mandatory, it could be an interesting reading if you want to know more details or even other usages not related to the Serverless Workflows.
+
```

Comment:
is there a room for what's next here?

Reply:
Not sure. This is the real end-road. People here already know about the services orchestration using OpenAPI. I think it's fine as it is unless @wmedvede has another idea.

Reply:
hahaha, this is sad.
I've just added the what's next section